controller/volume_controller.go

```diff
@@ -3008,6 +3095,22 @@ func (vc *VolumeController) updateRecurringJobs(v *longhorn.Volume) (err error)
 	if err = datastore.FixupRecurringJob(v); err != nil {
 		return err
 	}
+
+	// DR volume will not restore volume recurring jobs/groups configuration.
+	if v.Spec.FromBackup != "" && !v.Spec.Standby {
+		if !v.Status.RestoreInitiated {
+			RestoreRecurringJobs, err := vc.ds.GetSettingAsBool(types.SettingNameRestoreVolumeRecurringJobs)
+			if err != nil {
+				vc.logger.WithError(err).Warnf("Failed to get %v setting", types.SettingNameRestoreVolumeRecurringJobs)
+			}
+			if v.Spec.RestoreVolumeRecurringJob == longhorn.RestoreVolumeRecurringJobEnabled || (v.Spec.RestoreVolumeRecurringJob != longhorn.RestoreVolumeRecurringJobDisabled && RestoreRecurringJobs) {
```

Comment:
Do we still need this?

Reply:
Yes, User could decide to restore recurring jobs or not for a specific backup restoration to override the global setting.

Reply:
OK. Then it's better to use a spec field to control it, which is similar to field `volume.Spec.DataLocality`

Reply:
Yes, it is at a spec field `v.Spec.RestoreVolumeRecurringJob`.
I think you meant another flag `RestoringVolumeRecurringJob` in `volume.status` that had just been removed.